File: lista java/at11.java

abstract class Funcionario {
    protected String nome;

    public Funcionario(String nome) {
        this.nome = nome;
    }

    
    public abstract double calcularSalario();
}


class FuncionarioHorista extends Funcionario {
    private int horasTrabalhadas;
    private double salarioHora;

    public FuncionarioHorista(String nome, int horasTrabalhadas, double salarioHora) {
        super(nome);
        this.horasTrabalhadas = horasTrabalhadas;
        this.salarioHora = salarioHora;
    }

    @Override
    public double calcularSalario() {
        return horasTrabalhadas * salarioHora;
    }
}


class FuncionarioAssalariado extends Funcionario {
    private double salarioFixo;

    public FuncionarioAssalariado(String nome, double salarioFixo) {
        super(nome);
        this.salarioFixo = salarioFixo;
    }

    @Override
    public double calcularSalario() {
        return salarioFixo;
    }
}

public class Main {
    public static void main(String[] args) {
        Funcionario horista = new FuncionarioHorista("Jose", 160, 50);  
        Funcionario assalariado = new FuncionarioAssalariado("ROBERTA", 5000);  

        System.out.println("Salário de " + horista.nome + ": R$" + horista.calcularSalario());
        System.out.println("Salário de " + assalariado.nome + ": R$" + assalariado.calcularSalario());
    }
}

File: lista python/at11.py
from abc import ABC, abstractmethod


class Funcionario(ABC):
    def __init__(self, nome):
        self.nome = nome

    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, nome, horas_trabalhadas, salario_hora):
        super().__init__(nome)
        self.horas_trabalhadas = horas_trabalhadas
        self.salario_hora = salario_hora

    def calcular_salario(self):
        return self.horas_trabalhadas * self.salario_hora


class FuncionarioAssalariado(Funcionario):
    def __init__(self, nome, salario_fixo):
        super().__init__(nome)
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo


horista = FuncionarioHorista("Jose", 320, 100)  
assalariado = FuncionarioAssalariado("Roberta", 10000)  

print(f"Salário de {horista.nome}: R${horista.calcular_salario():.2f}")
print(f"Salário de {assalariado.nome}: R${assalariado.calcular_salario():.2f}")
